improver/calibration/rainforest_calibration.py

```diff
@@ -126,6 +144,424 @@ def __init__(self, model_config_dict: dict, threads: int):
                 for file in lightgbm_model_filenames
             ]
 
+    def _check_num_features(self, features: CubeList) -> None:
+        """Check that the correct number of features has been passed into the model.
+
+        Args:
+            features:
+                Cubelist containing feature variables.
+        """
+        from lightgbm import Booster
+
+        sample_tree_model = self.tree_models[0]
+
+        if isinstance(sample_tree_model, Booster):
+            expected_num_features = sample_tree_model.num_feature()
+        elif self.treelite_enabled:
+            from treelite_runtime import Predictor
+
+            if isinstance(sample_tree_model, Predictor):
+                expected_num_features = sample_tree_model.num_feature
+
+        if expected_num_features != len(features):
+            raise ValueError(
+                "Number of expected features does not match number of feature cubes."
+            )
+
+    def _align_feature_variables(
+        self, feature_cubes: CubeList, forecast_cube: Cube
+    ) -> Tuple[CubeList, Cube]:
+        """Ensure that feature cubes have consistent dimension coordinates. If
+        realization dimension present in any cube, all cubes lacking this dimension
+        will have realization dimension added and broadcast along this new dimension.
+
```

Comment:
It might be good to note in the docstring that this function is intended to be used for features that are not part of the ensemble NWP prediction (e.g. solar radiation).

Reply:
Good suggestion. I've extended this docstring to explain why some fields may not have a realization dimension. I've also added a comment as to how deterministic models are handled.